common/djangoapps/track/views/tests/__init__.py

```diff
@@ -0,0 +1,110 @@
+)
+class SegmentIOTrackingTestCaseBase(EventTrackingTestCase):
+    """
+    Base class for tests that test the processing of Segment events.
+    """
+
+    def setUp(self):
+        super(SegmentIOTrackingTestCaseBase, self).setUp()
+        self.maxDiff = None  # pylint: disable=invalid-name
+        self.request_factory = RequestFactory()
+
+    def create_request(self, key=None, **kwargs):
+        """Create a fake request that emulates a request from the Segment servers to ours"""
+        if key is None:
+            key = SEGMENTIO_TEST_SECRET
+
+        request = self.request_factory.post(SEGMENTIO_TEST_ENDPOINT + "?key=" + key, **kwargs)
+        if 'data' in kwargs:
+            request.json = json.loads(kwargs['data'])
+
+        return request
+
+    def post_segmentio_event(self, **kwargs):
+        """Post a fake Segment event to the view that processes it"""
+        request = self.create_request(
+            data=self.create_segmentio_event_json(**kwargs),
+            content_type='application/json'
+        )
+        segmentio.track_segmentio_event(request)
+
+    def create_segmentio_event(self, **kwargs):
+        """Populate a fake Segment event with data of interest"""
+        action = kwargs.get('action', 'Track')
+        sample_event = {
+            "userId": kwargs.get('user_id', SEGMENTIO_TEST_USER_ID),
+            "event": "Did something",
+            "properties": {
+                'name': kwargs.get('name', str(sentinel.name)),
+                'data': kwargs.get('data', {}),
+                'context': {
+                    'course_id': kwargs.get('course_id') or '',
+                    'app_name': 'edx.mobile.android',
+                }
+            },
+            "channel": 'server',
+            "context": {
+                "library": {
+                    "name": kwargs.get('library_name', 'test-app'),
+                    "version": "unknown"
+                },
+                "app": {
+                    "version": "1.0.1",
+                },
+                'userAgent': str(sentinel.user_agent),
+            },
+            "receivedAt": "2014-08-27T16:33:39.100Z",
+            "timestamp": "2014-08-27T16:33:39.215Z",
+            "type": action.lower(),
+            "projectId": "u0j33yjkr8",
+            "messageId": "qy52hwp4",
+            "version": 2,
+            "integrations": {},
+            "options": {
+                "library": "unknown",
+                "providers": {}
+            },
+            "action": action
+        }
+
+        if 'context' in kwargs:
+            sample_event['properties']['context'].update(kwargs['context'])
+        if 'label' in kwargs:
+            sample_event['properties']['context']['label'] = kwargs['label']
+
+        return sample_event
+
+    def create_segmentio_event_json(self, **kwargs):
+        """Return a json string containing a fake Segment event"""
+        return json.dumps(self.create_segmentio_event(**kwargs))
```

Comment:
Tiny nit: Was there a specific reason to put this code into `__init__.py` rather than a new file (`utils.py`, `helpers.py`, `segment_base.py`, etc.)? We can leave it if there is, I'm mostly just curious.

I ask because sometimes when I'm browsing a project, I forget to look inside __init__; a specifically-named file makes it impossible to miss.